modules/mono/glue/GodotSharp/GodotSharp/Core/NativeInterop/VariantUtils.cs

```diff
@@ -235,13 +235,28 @@ public static godot_variant CreateFromPackedColorArray(Span<Color> from)
         }
 
         public static godot_variant CreateFromSystemArrayOfStringName(Span<StringName> from)
-            => CreateFromArray(new Collections.Array(from));
+        {
+            if (from == null)
```

Comment:
This looks weird to me, Spans are value types so they can't be null. The only reason why they can be compared to null is because there is an implicit conversion from arrays to spans and the `null` literal can be converted to array so basically this is doing this:

```csharp
if (from == (Span<StringName>)((StringName[])null))
    return default;
```

Which means this also returns a default `Variant` for empty arrays and I'm not sure we want that.

Reply:
It's ok because for **empty** arrays, the span won't be null (as its `_pointer` points to that array). So, it returns default `Variant` only when the array is **null**.

Reply:
That's true, I was thinking whether we should allow creating `Godot.Array` from null but on second thought I think the changes in this PR are more _correct_.